components/script/dom/trackevent.rs

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::audiotrack::AudioTrack;
+use crate::dom::bindings::codegen::Bindings::EventBinding::EventBinding::EventMethods;
+use crate::dom::bindings::codegen::Bindings::TrackEventBinding;
+use crate::dom::bindings::codegen::Bindings::TrackEventBinding::TrackEventMethods;
+use crate::dom::bindings::codegen::UnionTypes::VideoTrackOrAudioTrackOrTextTrack;
+use crate::dom::bindings::error::Fallible;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::Dom;
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::event::Event;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::texttrack::TextTrack;
+use crate::dom::videotrack::VideoTrack;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_atoms::Atom;
+
+#[must_root]
+#[derive(JSTraceable, MallocSizeOf)]
```

Comment:
I am not very familiar with how/where to define the MallocSizeOf trait or ignored attribute as well as must_root/unrooted_must_root. I assumed here that since a new enum is created it should define MallocSizeOf or should the MediaTrack in TrackEvent still have the ignore_malloc_of_size above it?

Reply:
Deriving as often as possible is the best choice.